### DISCUSSION TITLE
Rate limits for database inserts per second

### DISCUSSION THREAD

**Discussion by tobbbe** [supabase/supabase#3651] [Questions] [answered]
Hello!
Is there any rate limits per second for database inserts? Either from Supabase or Postgres

**Comment by awalias** [comment, marked as answer]
we don't have any rate limits in place right now for the CRUD API, you can pretty much go as hard as you want since each database has it's own dedicated resources (which can be increased on the paid tiers)

I ran some benchmarks this morning and the CRUD API is capable of sustaining up to 1200 reads per second for the free tier, and up to 1000 inserts per second

soon you will be able to configure rate limiting yourself from the dashboard, but as of today this is a work in progress

**Reply by tobbbe** [reply]
WOW. Amazing, thank you! 😊

**Reply by cguagenti** [reply]
This is insane! I'm going all-in with Supabase for side projects and building some enterprise solutions as well. You're amazing guys!

**Reply by nmndwivedi** [reply]
Hi @awalias . Has the rate limiting from dashboard feature been dropped?

**Reply by daveycodez** [reply]
> we don't have any rate limits in place right now for the CRUD API, you can pretty much go as hard as you want since each database has it's own dedicated resources (which can be increased on the paid tiers)
> 
> I ran some benchmarks this morning and the CRUD API is capable of sustaining up to 1200 reads per second for the free tier, and up to 1000 inserts per second
> 
> soon you will be able to configure rate limiting yourself from the dashboard, but as of today this is a work in progress

So that was a lie

**Comment by oripka** [comment]
Any progress on this?

**Comment by deepak-coding-art** [comment]
Is it in place for the API and edge functions

**Comment by daveycodez** [comment]
1200 reads per second I wonder what that will do to your bill xD you guys are scammers.

**Comment by roo12312** [comment]
any progress?

**Comment by MrVibe** [comment]
Need throttle , otherwise I will always have billing tensions.
